src/app/components/mno-product-selector/mno-product-selector.controller.coffee

```diff
@@ -80,17 +80,22 @@
       ).finally(-> $ctrl.isLoadingProducts = false)
 
     # Filter products by name or category
+    filterProducts = ->
+      if $ctrl.searchTerm?.length > 0
+        term = $ctrl.searchTerm.toLowerCase()
+        $ctrl.filteredProducts = (product for product in $ctrl.products when product.name?.toLowerCase().indexOf(term) > -1)
+      else if $ctrl.selectedCategory?.length > 0
```

Comment:
How about `!!$ctrl.selectedCategory`?

Reply:
I'm personally more of a fan of ".length > 0" as it expresses the semantics check more clearly to readers.  However as the code does use !! more frequently, I'd rather be consistent.  I'll make the change.

Reply:
Good point, leave it like this. I'll push it for testing.

Reply:
Ok :smile: